src/dmd/mars.d

```diff
@@ -1361,13 +1361,25 @@ void addDefaultVersionIdentifiers()
     static if (TARGET.Windows)
     {
         if (global.params.mscoff)
+        {
             VersionCondition.addPredefinedGlobalIdent("CRuntime_Microsoft");
+            VersionCondition.addPredefinedGlobalIdent("CppRuntime_Microsoft");
+        }
         else
+        {
             VersionCondition.addPredefinedGlobalIdent("CRuntime_DigitalMars");
+            VersionCondition.addPredefinedGlobalIdent("CppRuntime_DigitalMars");
+        }
     }
     else static if (TARGET.Linux)
     {
         VersionCondition.addPredefinedGlobalIdent("CRuntime_Glibc");
+        VersionCondition.addPredefinedGlobalIdent("CppRuntime_StdCpp");
+    }
+    else static if (TARGET.OSX || TARGET.FreeBSD)
+    {
+        // why u no CRuntime guys?
+        VersionCondition.addPredefinedGlobalIdent("CppRuntime_LibCpp");
```

Comment:
As far as I could tell, these are the only guys that use clang as default toolchain...

Reply:
@jacob-carlborg Does this look correct to you?

Reply:
Probably not on macos, but certainly on linux, and even windows (mingw/cygwin). Toolchains could be built to target all sorts of combinations. LDC/GDC need the freedom to define these however the platform desires.

Reply:
I'm specifically talking about macOS here. I'll let someone else decide if this is necessary or not.

Reply:
I don't want to be writing:
```D
version(CppRuntime_LibCpp)
  version = CppRuntime_LibCpp_or_OSX;
else version (OSX)
  version = CppRuntime_LibCpp_or_OSX;
version (CppRuntime_LibCpp_or_OSX)
{
  // libc++ code...
}
```

Reply:
You would be writing:

```d
version (CppRuntime_StdCpp)
else version (OSX)
else version (FreeBSD)
```

That's how it's done for the C standard library: https://github.com/dlang/druntime/blob/master/src/core/stdc/locale.d#L60-L137.

Reply:
What about using `CppRuntime_GCC` and `CppRuntime_Clang`?

If I were to review code with `StdCpp` and `LibCpp` I would certainly have a hard time remembering the difference.

Reply:
> I'm specifically talking about macOS here.

C++ STL is tied to the compiler toolchain, not the OS.

Reply:
Fine. What about the rest of the platforms?

Reply:
Rest of the platforms accounted.

Reply:
>  That's how it's done for the C standard library: https://github.com/dlang/druntime/blob/master/src/core/stdc/locale.d#L60-L137.

Speaking of which, why don't we just add `CRuntime_OSX` and `CRuntime_BSD`?

I take it that FreeBSD, DragonFlyBSD, OpenBSD, ..., InsertNameHereBSD all share the same libc away.